pylib_3.9.7/df/AntibodyStructurePrediction.py

```diff
@@ -0,0 +1,223 @@
+                        model_number = model_number[:-row_multiplier]
+
+                        error_caught = True
+                    finally:
+                        if os.path.exists(pdb_filename):
+                            os.remove(pdb_filename)
+                        if error_caught:
+                            continue
+            else:
+                pdb_filename = '_'.join([ab_id, 'predicted.pdb'])
+                error_caught = False
+                try:
+                    antibody.save(pdb_filename)
+
+                    # open the file, compress, and encode it
+                    with open(pdb_filename, 'r', encoding='utf-8') as pdb_file:
+                        pdb_data = pdb_file.read()
+                        pdb_zip = gzip.compress(pdb_data.encode())
+                        pdb_enc = base64.b64encode(pdb_zip).decode('utf8')
+                        embedded_structures.append(pdb_enc)
+                except Exception as ex:
+                    notifications.append(Notification(level=NotificationLevel.ERROR,
+                                                      title='Antibody Structure Prediction',
+                                                      summary=f'Error saving ID {ab_id}',
+                                                      details=f'{ex.__class__} - {ex}'))
+
+                    # remove list elements for this broken item
+                    ids = ids[:-row_multiplier]
+                    orig_seq = orig_seq[:-row_multiplier]
+                    heavy_chain_seq = heavy_chain_seq[:-row_multiplier]
+                    light_chain_seq = light_chain_seq[:-row_multiplier]
+                    HL_concat_seq = HL_concat_seq[:-row_multiplier]
+                    model_rank = model_rank[:-row_multiplier]
+                    model_number = model_number[:-row_multiplier]
+
+                    error_caught = True
+                finally:
+                    if os.path.exists(pdb_filename):
+                        os.remove(pdb_filename)
+                    if error_caught:
+                        continue
+
+        # antibody numbering or not
+        # assume not, and set standard values
+        HL_chain_values = HL_concat_seq
+        HL_chain_props = {}
+        HL_chain_content_type = 'chemical/x-sequence'
+        HL_chain_data_type = DataType.STRING
+
+        if do_numbering:
+            try:
+                HL_chain_values, HL_chain_props = self._antibody_numbering([string_to_sequence(s, index)
+                                                                            for index, s in enumerate(HL_concat_seq)],
+                                                                           num_scheme, cdr_def)
+                HL_chain_content_type = 'chemical/x-genbank'
+                HL_chain_data_type = DataType.BINARY
+            except:
+                notifications.append(Notification(level=NotificationLevel.ERROR,
```

Comment:
Like the notification!

Reply:
Yeah, that bit I put in for BLAST Web Search is already coming in handy!  🥳